### ISSUE TITLE
Determine future of CI for PH repos

### ISSUE THREAD

**Issue by ZoeLeBlanc** [programminghistorian/jekyll#2016]
In November 2020, Travis CI (our current CI build operator) switched their pricing plan and we no longer had access to free CI through them. We've finally run out of credits on February 1 2021, and while we're hoping to get some access to free OSS credits through their support team, I want to start a ticket to determine if we need to move off of Travis CI to something that's free for our CI needs.

Mostly I'm concerned that Travis CI is requiring us to request OSS credits and that in the future they might revoke these or it might become a burden to have to request more every three months or so. Furthermore their pricing is a bit confusing but seems high for our needs (we would either have to pay $15 for 25000 credits or $69/month - tbh not sure how these pricing schemes interact). 

So here are some blog posts from other OSS maintainers about their responses to Travis's shift.
- https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works
- https://www.reddit.com/r/haskell/comments/jvbs49/psa_maintainers_should_think_about_migrating_off/

One thing we might consider is looking into CI through Github Actions. I have zero knowledge or experience doing this, but it seems to be a popular and free alternative.

**Comment by ZoeLeBlanc** [comment]
So some good news... it looks like we should be able to do everything build wise with Github Actions and remove our dependency on Travis completely 🎉 ! 

I've got github actions working for the search-index repo and eventually want to have it that any push into master triggers that repo as well. But first I need to get this repo working with github actions. This will probably break everything for a bit but I'm confident I can get it working eventually 😅

**Comment by ZoeLeBlanc** [comment]
Alrighty looks like we have a new system in place and are officially free of Travis 🍾 🥳 !

Before we close the issue here's some things we'll need to do to finalize the change:

- [x] Decide if we want to continue deploying Netlify through the service integration or on Github Actions (I'm leaning towards the former but want to double check with the @programminghistorian/technical-team )
	- [x] ~If Github Actions, remove @drjwbaker bank details from Netlify and close account.~
- [x] Comment the new build.yml files in `jekyll` and `search-index`
-  Remove the build.sh file (decided to keep it)
- [x] Remove the travis.yml file and delete travis from our service integrations (I would recommend we wait a few weeks to ensure that this new process is working)
- [x] Update our documentation in the wiki for Github Actions. Maybe we can add a retired technical documentation for the Travis specific content?
- Write a blog post about our shift to Github Actions (to both record some of this process and as a resource for other jekyll sites since a lot of people will be hitting this issue in the next year) (decided to write this on my personal blog forthcoming)

Think that's it for now but excited that this actually worked 👍

**Comment by drjwbaker** [comment]
Amazing job. I've added a sub-action above *if* you decide to ditch Netlify. I should add, I'm not suggesting you make a decision on grounds of cost. I merely want to ensure my bank details aren't lingering on a service we are no longer using :)

**Comment by walshbr** [comment]
@ZoeLeBlanc - are you sure that deploying netlify in github actions doesn't require the bank account? It's still using netlify, after all, and it still spins up on their servers. Is it linked to our account on there at all? I assumed it was just using the same netlify account but poking it from a different direction.

**Comment by ZoeLeBlanc** [comment]
@drjwbaker @walshbr sorry if that was confusing above, but yes we still need to pay for Netlify. In both approaches, everything is secure (if use Github Actions then we use repository secret tokens), but it really comes down to whether we prefer the UI of netlify or the UI of Github Actions. Also I think the netlify URLs are nicer if we deploy through their deploy-preview option, which might be available through Github Actions but I never saw it as an option.

So long story short we'll probably stay as is! Sorry again for the confusion!

**Comment by drjwbaker** [comment]
Thanks for explaining. No confusion here!
